Expand position-area CB to include in-flow scrollable area for ICB

fantasai opened the pull request:
#### 6d342d3c33b948e336800d85c61561cc94019730
<pre>
Expand position-area CB to include in-flow scrollable area for ICB
<a href="https://bugs.webkit.org/show_bug.cgi?id=299950">https://bugs.webkit.org/show_bug.cgi?id=299950</a>
<a href="https://rdar.apple.com/161741583">rdar://161741583</a>

Reviewed by Alan Baradlay.

This is not a complete fix for the spec requirement to cover the scrollable
area of scrollable containing blocks, but it fixes the most problematic case:
typical overflow of the initial containing block.

See <a href="https://bugs.webkit.org/show_bug.cgi?id=291864">https://bugs.webkit.org/show_bug.cgi?id=291864</a> for more complete follow-up.

Tests: imported/w3c/web-platform-tests/css/css-anchor-position/position-area-overflow-icb-001.html
       imported/w3c/web-platform-tests/css/css-anchor-position/position-area-overflow-icb-002.html
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-overflow-icb-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-overflow-icb-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-overflow-icb-002-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-overflow-icb-002.html: Added.

Add some basic reftests for this case.

* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-scrolling-003.tentative-expected.txt:

Mark test as failing. The test is wrong in any case, but also not worth fixing
until we have a more complete fix for this issue.

* Source/WebCore/rendering/PositionedLayoutConstraints.cpp:
(WebCore::PositionedLayoutConstraints::captureScrollableArea):
* Source/WebCore/rendering/PositionedLayoutConstraints.h:

Expand the position-area ICB to cover the area of an in-flow root element.

Canonical link: <a href="https://commits.webkit.org/300921@main">https://commits.webkit.org/300921@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d80be718c95fba3da5a73bf084379021e5305fe4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124306 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43988 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34716 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131144 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/76369 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/3d428fe6-57c2-49d2-b85b-42a508e5078e) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126183 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44733 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52589 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94555 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/76369 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/911ca6c6-607d-4c54-9d26-48b4a2e42bf5) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127260 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35649 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111181 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75143 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d8a1a0cd-393d-4943-9176-2531c221f734) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34590 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29341 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74622 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105392 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29564 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133811 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51212 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39049 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103035 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51607 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107400 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102831 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26172 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48194 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26445 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48151 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51076 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/56862 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50513 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53869 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52188 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d80be718c95fba3da5a73bf084379021e5305fe4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51718

fantasai commented:
WPT PR at https://github.com/web-platform-tests/wpt/pull/55217

fantasai committed:
Expand position-area CB to include in-flow scrollable area for ICB
https://bugs.webkit.org/show_bug.cgi?id=299950
rdar://161741583

Reviewed by Alan Baradlay.

This is not a complete fix for the spec requirement to cover the scrollable
area of scrollable containing blocks, but it fixes the most problematic case:
typical overflow of the initial containing block.

See https://bugs.webkit.org/show_bug.cgi?id=291864 for more complete follow-up.

Tests: imported/w3c/web-platform-tests/css/css-anchor-position/position-area-overflow-icb-001.html
       imported/w3c/web-platform-tests/css/css-anchor-position/position-area-overflow-icb-002.html
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-overflow-icb-001-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-overflow-icb-001.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-overflow-icb-002-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-overflow-icb-002.html: Added.

Add some basic reftests for this case.

* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/position-area-scrolling-003.tentative-expected.txt:

Mark test as failing. The test is wrong in any case, but also not worth fixing
until we have a more complete fix for this issue.

* Source/WebCore/rendering/PositionedLayoutConstraints.cpp:
(WebCore::PositionedLayoutConstraints::captureScrollableArea):
* Source/WebCore/rendering/PositionedLayoutConstraints.h:

Expand the position-area ICB to cover the area of an in-flow root element.

Canonical link: https://commits.webkit.org/300921@main

webkit-commit-queue commented:
Committed 300921@main (6d342d3c33b9): <https://commits.webkit.org/300921@main>

Reviewed commits have been landed. Closing PR #51718 and removing active labels.
